app/models/service_template.rb

```diff
@@ -435,6 +437,10 @@ def provision_workflow(user, dialog_options = nil, request_options = {})
     end
   end
 
+  def initialize_copy(_orig)
+    super.tap { |obj| obj.update_attributes(:guid => nil, :created_at => Time.zone.now, :updated_at => Time.zone.now) }
```

Comment:
I understand setting `guid` to '`nil` but won't `created_at` / `updated_at` be set by rails?  Or are they required here for some reason?

Reply:
``` ActiveRecord::StatementInvalid:
        PG::NotNullViolation: ERROR:  null value in column "created_at" violates not-null constraint``` without them.

Reply:
something must have gone wrong. This is the first time I saw `update_attributes` requires such fields. Was `obj` already saved before `update_attributes`?